### PR TITLE
Expose the ability to stop a server

### PR DESCRIPTION
**Pull request by nkoenig**
Signed-off-by: Nate Koenig <nate@openrobotics.org>

# 🎉 New feature

## Summary

Exposes `ServerPrivate::Stop` through a new `Server::Stop` function. This is useful when using Gazebo as a library.
 

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [x] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: src/Server_TEST.cc [hunk 1]
```diff
@@ -1172,6 +1172,32 @@ TEST_P(ServerFixture, ResolveResourcePaths)
         "include_nested", "model.sdf"), true);
 }
 
+/////////////////////////////////////////////////
+TEST_P(ServerFixture, Stop)
+{
+  // Start server
+  ServerConfig serverConfig;
+  serverConfig.SetUpdateRate(10000);
+  serverConfig.SetSdfFile(std::string(PROJECT_SOURCE_PATH) +
```
Comment (ahcorde) [inline src/Server_TEST.cc:1181]: `joinpaths`

**Review by ahcorde — CHANGES_REQUESTED** [review]

File: src/Server_TEST.cc (hunk 1 above)
Comment (nkoenig) [reply src/Server_TEST.cc]: Done in [40c37d2](https://github.com/gazebosim/gz-sim/pull/1551/commits/40c37d295c74c67bbcc2bbe7472a584f38a64a00)

**Review by chapulina — APPROVED** [review]

File: src/Server_TEST.cc [hunk 2]
```diff
@@ -231,8 +231,8 @@ TEST_P(ServerFixture, IGN_UTILS_TEST_DISABLED_ON_WIN32(ServerConfigRealPlugin))
   // Start server
   ServerConfig serverConfig;
   serverConfig.SetUpdateRate(10000);
-  serverConfig.SetSdfFile(std::string(PROJECT_SOURCE_PATH) +
-      "/test/worlds/shapes.sdf");
+  serverConfig.SetSdfFile(common::joinPaths(PROJECT_SOURCE_PATH,
+      "/test/worlds/shapes.sdf"));
```
Comment (chapulina) [inline src/Server_TEST.cc:235]: Ahh actually this needs to be

```suggestion
      "test", "worlds", "shapes.sdf"));
```

**Review by chapulina — CHANGES_REQUESTED** [review]

**Review by chapulina — APPROVED** [review]
LGTM when CI comes back happy

**Pull request merged**